WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerFragment.java

```diff
@@ -679,4 +684,15 @@ private void trackSelectedOtherSourceEvents(AnalyticsTracker.Stat stat, boolean
         properties.put("is_video", isVideo);
         AnalyticsTracker.track(stat, properties);
     }
+
+    private ArrayList<EditImageData.InputData> createListOfEditImageInputData(List<Uri> uris) {
+        ArrayList<EditImageData.InputData> inputData = new ArrayList<>(uris.size());
+        for (Uri uri : uris) {
+            String stringUri = uri.toString();
+            String outputFileExtension = MimeTypeMap.getFileExtensionFromUrl(stringUri);
```

Comment:
`MimeTypeMap.getFileExtensionFromUrl(stringUri)` is returning a blank value here and results in transparency loss for png images. We can probably create a function like this that checks for `SCHEME_CONTENT` when determining the file extension:

```
   public String getFileExtension(Uri uri) {
        String fileExtension;
        if (uri.getScheme() != null && uri.getScheme().equals(ContentResolver.SCHEME_CONTENT)) {
            ContentResolver cr = requireContext().getContentResolver();
            String mimeType = cr.getType(uri);
            fileExtension = MimeTypeMap.getSingleton().getExtensionFromMimeType(mimeType);
        } else {
            fileExtension = MimeTypeMap.getFileExtensionFromUrl(uri.toString());
        }
        return fileExtension;
    }
```

Reply:
Nice catch! ;) Fixed in 39074c2c389829057318343b297a3d7d364d469b